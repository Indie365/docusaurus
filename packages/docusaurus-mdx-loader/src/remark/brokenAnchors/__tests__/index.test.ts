/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import remark2rehype from 'remark-rehype';
import stringify from 'rehype-stringify';
import vfile from 'to-vfile';
import plugin from '../index';
import admonition from '../../admonitions';
import type {WebpackCompilerName} from '@docusaurus/utils';

const processFixture = async (
  name: string,
  {compilerName}: {compilerName: WebpackCompilerName},
) => {
  const {remark} = await import('remark');

  const filePath = path.join(__dirname, '__fixtures__', `${name}.md`);
  const file = await vfile.read(filePath);
  file.data.compilerName = compilerName;

  const result = await remark()
    .use(admonition)
    .use(plugin)
    .use(remark2rehype)
    .use(stringify)
    .process(file);

  return result.value;
};

describe('directives remark plugin - client compiler', () => {
  const consoleMock = jest.spyOn(console, 'warn').mockImplementation(() => {});
  beforeEach(() => jest.clearAllMocks());

  const options = {compilerName: 'client'} as const;

  it('default behavior for hello file', async () => {
    const result = await processFixture('hello', options);
    expect(result).toMatchSnapshot('result');
    expect(consoleMock).toHaveBeenCalledTimes(1);
    expect(consoleMock.mock.calls).toMatchSnapshot('console');
  });

  it('default behavior for world file', async () => {
    const result = await processFixture('world', options);
    expect(result).toMatchSnapshot('result');
    expect(consoleMock).toHaveBeenCalledTimes(1);
    expect(consoleMock.mock.calls).toMatchSnapshot('console');
  });
});
