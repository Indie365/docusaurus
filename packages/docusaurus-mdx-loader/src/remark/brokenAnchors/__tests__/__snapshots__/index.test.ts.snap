// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directives remark plugin - client compiler default behavior for hello file: console 1`] = `
[
  [
    "[WARNING] Docusaurus found 3 broken anchor(s) in file "<PROJECT_ROOT>/packages/docusaurus-mdx-loader/src/remark/brokenAnchors/__tests__/__fixtures__/anchors.md":
custom-id, notvalidinheading, notvalid",
  ],
]
`;

exports[`directives remark plugin - client compiler default behavior for hello file: result 1`] = `
"<h1><a href="#found">Hello</a></h1>
<h2><a href="#hello">found</a></h2>
<h2><a href="#title-with-space">title link with space</a></h2>
<p><a href="#hello">Found</a></p>
<h3>title with space</h3>
<p><a href="#title-with-space">title with space</a></p>
<h3>title</h3>
<p><a href="#title">title</a></p>
<p><a href="remoteAnchors.md#world">World found</a></p>
<p><a href="remoteAnchors.md">world</a></p>
<p><a href="./subfolder/remote.md#test">test</a></p>
<h2>test {#custom-id}</h2>
<p><a href="#custom-id">custom-id</a></p>
<h3><a href="#notvalidinheading">Not found</a></h3>
<p><a href="#notvalid">Not found</a></p>
<p><a href="remoteAnchors.md#worldnotvalid">World not found</a></p>
<p><a href="https://google.com">google</a></p>
<p><a href="#emptylink"></a></p>
<h6></h6>"
`;

exports[`directives remark plugin - client compiler default behavior for world file: console 1`] = `[]`;

exports[`directives remark plugin - client compiler default behavior for world file: result 1`] = `"<h1>World</h1>"`;
